### DISCUSSION TITLE
Query Strings are applied to all requests in Laravel Octane

### DISCUSSION THREAD

**Discussion by sbrow** [livewire/livewire#3509] [7. Bugs (archived)] [answered]
I have two components, both rendered as full pages. One of them has a nullable property `$type` that exists in the query string.

When using Laravel Octane, the second I set `$type`, the query string appears on ALL of my Livewire pages, and I can't get rid of it.

i.e If I navigate to `/a?type=foo` and then navigate to `/b` I am redirected to `/b?type=foo` even though that page doesn't have a `type` property.

Here are some sample classes:
```php
class A extends \Livewire\Component {
    public ?string $type;

    protected $queryString = [ 'type' ];

    public function render() {
        return view('livewire.a');
    }
}
```
```php
class B extends \Livewire\Component {
    public function render() {
        return view('livewire.b');
    }
}
```

**Comment by michael-rubel** [comment]
The issue still exists. I've disabled all query string operations for a while in my projects. Looks like no one PRed to fix this actually, so I will probably dig deeper into the problem a bit later.

**Comment by joshhanley** [comment]
@sbrow and @michael-rubel I've just tried to replicate this issue and I couldn't. Any chance one of you could provide a small repo that demonstrates this issue?

Thanks!

**Reply by joshhanley** [reply]
Actually, don't worry, managed to replicate the issue!

**Reply by georgeboot** [reply]
@joshhanley any way I can help getting this fixed? Or are you already almost there?

**Reply by joshhanley** [reply]
@georgeboot thanks! Already have a solution for it, will hopefully be able to run it by Caleb in the next day or two and then I will get it implemented 🙂

**Reply by georgeboot** [reply]
Fixed in https://github.com/livewire/livewire/pull/3987

**Reply by joshhanley** [reply]
@georgeboot heh you beat me to it! 😁

Have to wait to see if the Octane PR gets merged into Octane first though before it can be marked as fixed 🙂

**Reply by georgeboot** [reply]
Yeah and alternatively, Livewire could supply the listener and people will have to add it to their Octane config.

**Reply by joshhanley** [reply]
It was merged! 😁

**Comment by joshhanley** [comment, marked as answer]
Both the Livewire livewire/livewire#3987 and laravel/octane#400 PR's have been merged, so this should be fixed in the next releases of Octane and Livewire.

Hope this helps!
